Src/PerformanceCollector/Shared/QuickPulseTelemetryModule.cs

```diff
@@ -0,0 +1,468 @@
+
+        private readonly object collectedSamplesLock = new object();
+        
+        private readonly Uri serviceUriDefault = new Uri("https://qps.com/api");
+
+        private readonly LinkedList<QuickPulseDataSample> collectedSamples = new LinkedList<QuickPulseDataSample>();
+
+        private string instrumentationKey;
+
+        private IQuickPulseServiceClient serviceClient;
+
+        private Timer collectionTimer;
+
+        private Timer stateTimer;
+
+        private QuickPulseTimeProvider timeProvider;
+
+        private QuickPulseTimings timings;
+
+        private bool isInitialized;
+
+        private QuickPulseCollectionTimeSlotManager collectionTimeSlotManager = null;
+
+        private IQuickPulseDataAccumulatorManager dataAccumulatorManager = null;
+
+        private IQuickPulseTelemetryProcessor telemetryProcessor = null;
+
+        private QuickPulseCollectionStateManager stateManager = null;
+
+        private IPerformanceCollector performanceCollector = null;
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QuickPulseTelemetryModule"/> class.
+        /// </summary>
+        public QuickPulseTelemetryModule()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QuickPulseTelemetryModule"/> class. Internal constructor for unit tests only.
+        /// </summary>
+        /// <param name="collectionTimeSlotManager">Collection time slot manager.</param>
+        /// <param name="dataAccumulatorManager">Data hub to sink QuickPulse data to.</param>
+        /// <param name="telemetryProcessor">Telemetry initializer to inspect telemetry stream.</param>
+        /// <param name="serviceClient">QPS service client.</param>
+        /// <param name="performanceCollector">Performance counter collector.</param>
+        /// <param name="timings">Timings for the module.</param>
+        internal QuickPulseTelemetryModule(
+            QuickPulseCollectionTimeSlotManager collectionTimeSlotManager,
+            QuickPulseDataAccumulatorManager dataAccumulatorManager,
+            IQuickPulseTelemetryProcessor telemetryProcessor,
+            IQuickPulseServiceClient serviceClient,
+            IPerformanceCollector performanceCollector,
+            QuickPulseTimings timings)
+            : this()
+        {
+            this.collectionTimeSlotManager = collectionTimeSlotManager;
+            this.dataAccumulatorManager = dataAccumulatorManager;
+            this.telemetryProcessor = telemetryProcessor;
+            this.serviceClient = serviceClient;
+            this.performanceCollector = performanceCollector;
+            this.timings = timings;
+        }
+
+        /// <summary>
+        /// Gets the QuickPulse service endpoint to send to.
+        /// </summary>
+        /// <remarks>Loaded from configuration.</remarks>
+        public string QuickPulseServiceEndpoint { get; private set; }
```

Comment:
So changing of endpoint in runtime wouldn't affect where we send telemetry to?
Why is it `private set`? Don't you want to configure it from `ApplicationInsights.config` file?